src/Controller/CRUDController.php

```diff
@@ -655,13 +622,9 @@ public function showAction($deprecatedId = null) // NEXT_MAJOR: Remove the unuse
         $fields = $this->admin->getShow();
         \assert($fields instanceof FieldDescriptionCollection);
 
-        // NEXT_MAJOR: replace deprecation with exception
         if (!\is_array($fields->getElements()) || 0 === $fields->count()) {
-            @trigger_error(
-                'Calling this method without implementing "configureShowFields"'
-                .' is not supported since sonata-project/admin-bundle 3.40.0'
-                .' and will no longer be possible in 4.0',
-                E_USER_DEPRECATED
+            throw new \RuntimeException(
+                'No fields defined. Did you forget to implement the "configureShowFields" method?'
```

Comment:
I don't know if this is completely right, but I didn't found this method declared in other interface:
```suggestion
                'No fields defined. Did you forget to implement `AdminExtensionInterface::configureShowFields()`?'
```

Reply:
Technically, it should say `AdminInterface::getShow`, but since everyone extends from `AbstractAdmin`, it refers to the `AbstractAdmin::configureShowFields` where you define your fields. The same happen to [createAction](https://github.com/sonata-project/SonataAdminBundle/blob/master/src/Controller/CRUDController.php#L538) or editAction.

Reply:
`AbstractAdmin::configureShowFields()` (and the others) are not abstract, so they aren't forcing to override their definition :thinking:

Reply:
Yes, you're right, they are not forced to override it, only when they want to access to the action. In the docs [we are telling people to extend from `AbstractAdmin`](https://sonata-project.org/bundles/admin/3-x/doc/getting_started/creating_an_admin.html#step-1-create-an-admin-class) and in this case, for the `show` action, [we are telling them to override `configureShowFields`](https://sonata-project.org/bundles/admin/3-x/doc/reference/action_show.html#the-show-action), so I think the developer would know what `configureShowFields` method means, maybe I should change it to something like:
`No fields defined, Did you forget to override the configureShowFields  in your Admin class?`.

Reply:
It's fine for me in the context of this PR, since IMO, this situation deserves a more exhaustive analysis.